pkg/schedule/scatter/region_scatterer.go

```diff
@@ -340,9 +326,15 @@ func (r *RegionScatterer) scatterRegion(region *core.RegionInfo, group string, s
 				// It is both sourcePeer and targetPeer itself, no need to select.
 				continue
 			}
+			sourceStore := r.cluster.GetStore(peer.GetStoreId())
+			if sourceStore == nil {
+				log.Error("failed to get the store", zap.Uint64("store-id", peer.GetStoreId()), errs.ZapError(errs.ErrGetSourceStore))
+				continue
+			}
+			filters[filterLen-1] = filter.NewPlacementSafeguard(r.name, r.cluster.GetSharedConfig(), r.cluster.GetBasicCluster(), r.cluster.GetRuleManager(), region, sourceStore, oldFit)
```

Comment:
But we don't change region info, so this filter can be used again.
Maybe u need to update `region`

Reply:
but the source store is different

Reply:
Get, but I think it's nice to put it in `selectNewPeer` like before

Reply:
good point，I will change it